packages/atomic/src/utils/accessibility-utils.ts

```diff
@@ -49,12 +51,79 @@ export class AriaLiveRegionController implements ReactiveController {
   }
 }
 
-//TODO: Reimplement to fit Lit
-export class FocusTargetController {
-  public setTarget(_el: HTMLElement) {}
-  public focusAfterSearch() {}
-  public focusOnNextTarget() {}
-  public async focus() {}
+export class FocusTargetController implements ReactiveController {
+  private bindings: AnyBindings;
+  private lastSearchId?: string;
+  private element?: HTMLElement;
+  private onFocusCallback?: Function;
+  private doFocusAfterSearch = false;
+  private doFocusOnNextTarget = false;
+
+  constructor(
+    private host: ReactiveControllerHost,
+    bindings: AnyBindings
+  ) {
+    this.host = host;
+    this.bindings = bindings;
+    this.host.addController(this);
+  }
+
+  public setTarget(el?: HTMLElement) {
+    if (!el) {
+      return;
+    }
+    this.element = el;
+    if (this.doFocusOnNextTarget) {
+      this.doFocusOnNextTarget = false;
+      this.focus();
+    }
+  }
+
+  public async focus() {
+    await defer();
```

Comment:
@fbeaudoincoveo do you know why we need a defer?

Reply:
I presume it's to wait for the next animation frame, but it's hard to say for sure.

I would keep it for now: I wouldn't want to risk introducing hard to debug regressions during the migration. This is something we could delve into in isolation later.

Reply:
```suggestion
    // Not sure why this is needed; should be investigated after Lit Migration (KIT-4235)
    await defer();
```